web/ui_utils.js

```diff
@@ -193,6 +193,11 @@ function watchScroll(viewAreaElement, callback, abortSignal = undefined) {
     useCapture: true,
     signal: abortSignal,
   });
+  abortSignal?.addEventListener(
+    "abort",
+    () => window.cancelAnimationFrame(rAF),
```

Comment:
Should we add an `if (rAF)` check around this given that `rAF` may be `null` at this point if the viewer is closed before any scroll happened? We usually do the same for `clearTimeout`; see e.g. https://github.com/mozilla/pdf.js/blob/2fbd61944b88d67a2df97ae8a3ac1fa86f7b4092/web/pdf_rendering_queue.js#L73 and https://github.com/mozilla/pdf.js/blob/2fbd61944b88d67a2df97ae8a3ac1fa86f7b4092/web/pdf_presentation_mode.js#L264

Reply:
I don't know if it how much matters here, since running `window.cancelAnimationFrame(null);` in e.g. the browser console doesn't throw?

Reply:
Hm, interesting. I just tried the same with `clearTimeout` and that also doesn't throw, so even though it's not documented it was probably taken into account for the implementation. In that case my comment can be disregarded.

Reply:
There is nothing in the spec about the type of the handle returned by `requestAnimationFrame` (like for `setTimeout` as far as I remember), so it should be possible to have whatever we want, only a valid handle will lead to the removal of the callback.
That said in the docs: https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelAnimationFrame
this function isn't supposed to throw.